$NetBSD: patch-ai,v 1.1 2006/07/14 14:57:13 rillig Exp $

Assume that if CMSG_FIRSTHDR is not defined, struct msghdr.msg_control
does not exist either. Needed for Solaris 8.

--- gnome-pty-helper/gnome-pty-helper.c.orig	2006-02-14 10:00:32.000000000 +0100
+++ gnome-pty-helper/gnome-pty-helper.c	2006-07-14 01:38:57.548708903 +0200
@@ -123,6 +123,7 @@ pass_fd (int client_fd, int fd)
 	msg.msg_iovlen     = 1;
 	msg.msg_name       = NULL;
 	msg.msg_namelen    = 0;
+#if defined(CMSG_FIRSTHDR)
 	msg.msg_control    = (caddr_t) cmptr;
 	msg.msg_controllen = CONTROLLEN;
 
@@ -130,6 +131,7 @@ pass_fd (int client_fd, int fd)
 	cmptr->cmsg_type  = SCM_RIGHTS;
 	cmptr->cmsg_len   = CONTROLLEN;
 	*(int *)CMSG_DATA (cmptr) = fd;
+#endif
 
 	if (sendmsg (client_fd, &msg, 0) != 1)
 		return -1;
